### PR TITLE
workflows: run lerna directly and fix master build + run build before tests locally

### LINKED ISSUE #444: Tests are disable and failing in master
when your github action runs:
https://github.com/spotify/backstage/runs/552952974?check_suite_focus=true

The tests don't run in master because you have --since origin/master
```
Run yarn test
yarn run v1.22.1
$ cross-env CI=true lerna run test --since origin/master -- --coverage
lerna notice cli v3.20.2
lerna info ci enabled
lerna notice filter changed since "origin/master"
lerna info Looking for changed packages since origin/master
lerna success run No packages found with the lifecycle script 'test'
Done in 0.58s.
```

If I remove --since origin/master to run all tests,  the tests are failing with:
```
lerna ERR! yarn run test --coverage stderr:
 FAIL  src/components/WelcomePage/ToggleFeatureFlagButton.test.tsx
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/components/WelcomePage/ToggleFeatureFlagButton.test.tsx:25:8 - error TS2307: Cannot find module '@backstage/core'.

    25 } from '@backstage/core';
              ~~~~~~~~~~~~~~~~~

 FAIL  src/plugin.test.ts
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/plugin.ts:17:30 - error TS2307: Cannot find module '@backstage/core'.

    17 import { createPlugin } from '@backstage/core';
                                    ~~~~~~~~~~~~~~~~~
    src/plugin.ts:22:14 - error TS7031: Binding element 'router' implicitly has an 'any' type.

    22   register({ router, featureFlags }) {
                    ~~~~~~
    src/plugin.ts:22:22 - error TS7031: Binding element 'featureFlags' implicitly has an 'any' type.

    22   register({ router, featureFlags }) {
                            ~~~~~~~~~~~~

 FAIL  src/components/WelcomePage/WelcomePage.test.tsx
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/components/WelcomePage/WelcomePage.test.tsx:26:8 - error TS2307: Cannot find module '@backstage/core'.

    26 } from '@backstage/core';
              ~~~~~~~~~~~~~~~~~

 FAIL  src/components/WelcomePage/ErrorButton.test.tsx
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/components/WelcomePage/ErrorButton.test.tsx:20:55 - error TS2307: Cannot find module '@backstage/core'.

    20 import { ApiRegistry, errorApiRef, ApiProvider } from '@backstage/core';
                                                             ~~~~~~~~~~~~~~~~~

Failed to collect coverage from /Users/557659/backstage/plugins/welcome/src/plugin.ts
ERROR: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/plugin.ts:17:30 - error TS2307: Cannot find module '@backstage/core'.

17 import { createPlugin } from '@backstage/core';
                                ~~~~~~~~~~~~~~~~~
src/plugin.ts:22:14 - error TS7031: Binding element 'router' implicitly has an 'any' type.

22   register({ router, featureFlags }) {
                ~~~~~~
src/plugin.ts:22:22 - error TS7031: Binding element 'featureFlags' implicitly has an 'any' type.

22   register({ router, featureFlags }) {
                        ~~~~~~~~~~~~
STACK: 
Failed to collect coverage from /Users/557659/backstage/plugins/welcome/src/components/Timer/Timer.tsx
ERROR: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/components/Timer/Timer.tsx:18:29 - error TS2307: Cannot find module '@backstage/core'.

18 import { HeaderLabel } from '@backstage/core';
                               ~~~~~~~~~~~~~~~~~
STACK: 
Failed to collect coverage from /Users/557659/backstage/plugins/welcome/src/components/WelcomePage/ToggleFeatureFlagButton.tsx
ERROR: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/components/WelcomePage/ToggleFeatureFlagButton.tsx:19:62 - error TS2307: Cannot find module '@backstage/core'.

19 import { FeatureFlagState, featureFlagsApiRef, useApi } from '@backstage/core';
                                                                ~~~~~~~~~~~~~~~~~
STACK: 
Failed to collect coverage from /Users/557659/backstage/plugins/welcome/src/components/WelcomePage/ErrorButton.tsx
ERROR: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/components/WelcomePage/ErrorButton.tsx:19:37 - error TS2307: Cannot find module '@backstage/core'.

19 import { errorApiRef, useApi } from '@backstage/core';
                                       ~~~~~~~~~~~~~~~~~
STACK: 
Failed to collect coverage from /Users/557659/backstage/plugins/welcome/src/components/WelcomePage/WelcomePage.tsx
ERROR: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/components/WelcomePage/WelcomePage.tsx:36:8 - error TS2307: Cannot find module '@backstage/core'.

36 } from '@backstage/core';
          ~~~~~~~~~~~~~~~~~
STACK: 
Test Suites: 4 failed, 4 total
Tests:       0 total
Snapshots:   0 total
Time:        39.516s
Ran all test suites.
Command 'web-scripts' exited with code 1
error Command failed with exit code 1.

lerna ERR! yarn run test --coverage exited 1 in '@backstage/plugin-welcome'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes #444

### CODE REVIEW

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -48,12 +48,21 @@ jobs:
...
           node-version: ${{ matrix.node-version }}
+
       - name: yarn install
         run: yarn install --frozen-lockfile
-      - run: yarn lint
-      - run: yarn build
-      - run: yarn test
+
+      - name: lint
+        run: yarn lerna -- run lint --since origin/master
```
Comment (freben) [inline .github/workflows/frontend.yml:56]: this one is `yarn lerna -- run lint`, the below is `yarn lerna run lint` - i'm not clear on what the double dashes would be for here. Same goes for all the places further down, it varies for some reason.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -47,16 +47,25 @@ jobs:
...
           registry-url: https://registry.npmjs.org/ # Needed for auth
+
       - name: yarn install
         run: yarn install --frozen-lockfile
-      - run: yarn lint
-      - run: yarn build
-      - run: yarn test
+
+      - name: lint
+        run: yarn lerna run lint
```
Comment (freben) [inline .github/workflows/master.yml:55]: not clear on why this isn't just `yarn lint`

**Review by freben — DISMISSED** [review]
Approving, but with questions

File: .github/workflows/frontend.yml (hunk 1 above)
Comment (Rugvip) [reply .github/workflows/frontend.yml:56]: The first set of `--` is for yarn, the second is for lerna. Not really needed everywhere but added it for consistency. Fixed some inconsistencies.

File: .github/workflows/master.yml (hunk 2 above)
Comment (Rugvip) [reply .github/workflows/master.yml]: `yarn lint` as actually supposed to be different, hadn't pushed the latest update :grin:

**Review by freben — APPROVED** [review]
Yeah this is clearer.

Still wondering if the workflow scripts is where we want to put the complexity. Could also have been exposed as more script directives in the root package.json. We could have a `test:ci` that does not build but does have `--coverage` always, and the master workflow adds just `--since` or whatever.

Maybe even clearer - `build` just triggers `build:local` (to be super explicit that that's what it is for), and then there's also `build:ci` as above.

Anyway, just food for thought, some rainy day we can keep tweaking that. Good job fixing this

**Comment by Rugvip** [comment]
Yeah agree that it can be improved further and `test:ci` prolly makes sense.

**Pull request merged**